docs/proposals/dynamic-kubelet-settings.md

```diff
@@ -0,0 +1,129 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+A proposal for improving Kubelet config by versioning it and making it manageable via the API server.
+
+## Motivation
+
+The Kubelet is currently configured via command-line flags. This is painful for a number of reasons:
+- It makes it difficult to change the way Kubelets are configured in a running cluster, because it is often tedious to change the Kubelet startup configuration.
```

Comment:
Ansible or salt can do that for you. You will have problems when you update kubelet configuration in apiserver which gets reflected in a kubelet vs. re-running ansible/running salt. With ansible, it is quote easy to skip configuration tasks (``skip-tags`` flag) once the cluster is deployed and the ansible is re-run. So the new kubelet configuration does not get touched (still, need to keep that in mind). With salt, not sure if you can do anything here unless you stop the salt, change salt (e.g. skip configuration) and start salt.

Saying, some folks can be running ansible/salt already. When you start configuring the kubelet dynamically by default, you can get into trouble.